esy/src/release.js

```diff
@@ -193,19 +193,30 @@
  *  `<binary_name>` calls out to ocaml again. See
  *  `<PACKAGE_NAME>_ENVIRONMENT_SOURCED_<binary_name>`
  *
+ *  @flow
  */
 
-var fs = require('fs');
-var path = require('path');
-var child_process = require('child_process');
+import * as fs from './lib/fs';
+import * as child_process from './lib/child_process';
+const os = require('os');
+const path = require('path');
 
-var storeVersion = '3.x.x';
+type ReleaseType = 'dev' | 'pack' | 'bin';
+
+type BuildReleaseConfig = {
+  type: ReleaseType,
+  version: string,
+  sandboxPath: string,
+};
+
+type ReleaseConfig = {
+  type: ReleaseType,
+  version: string,
+  origin: string,
+  forGithubLFS: boolean,
```

Comment:
BTW: I ended up not needing to support github large file storage. The trick was that I bundled up each individual prebuilt binary for each dependency into their own tar/gz each of which ends up being under the github limit.
- It was pretty hacky how I did that: In `sh`, I computed the final `PATH` and split it on `:` to find all the directories for all esy store transitive dependencies. This makes it so you can use the global store instead of creating a new empty build store every time you release (so slow).  I think `esy` should probably expose a library/command that will compute the set of transitive store directories for you without having to do hacky string manipulations like [I did](https://github.com/reasonml/reason-cli/blob/master/releaseUtils/release.js#L465)

Reply:
> I think esy should probably expose a library/command that will compute the set of transitive store directories

Already added in 83a1af4. It's to support `release` command at the moment, but I think we'd need a more structured db at some point.